api/scpca_portal/test/management/commands/test_create_portal_metadata.py

```diff
@@ -54,31 +52,56 @@ def load_test_data(self):
         self.assertEqual(Sample.objects.all().count(), SAMPLES_COUNT)
         self.assertEqual(Library.objects.all().count(), LIBRARIES_COUNT)
 
-    def test_create_portal_metadata(self):
+    # TODO: After PR #839 is merged into dev, add readme file format testing
+    def assertProjectReadmeContains(self, text, zip_file):
+        self.assertIn(text, zip_file.read(README_FILE).decode("utf-8"))
+
+    @patch("scpca_portal.management.commands.create_portal_metadata.s3.upload_output_file")
+    def test_create_portal_metadata(self, mock_upload_output_file):
+        # Set up the database for test
         self.load_test_data()
-        self.processor.create_portal_metadata(clean_up_output_data=False)
+        # Create the portal metadata computed file
+        self.processor.create_portal_metadata(clean_up_output_data=False, update_s3=True)
+
+        # Test the computed file
+        # Make sure the computed file is created and singular
+        computed_files = ComputedFile.objects.filter(portal_metadata_only=True)
+        self.assertEqual(computed_files.count(), 1)
+        computed_file = computed_files.first()
+        # Make sure the computed file size is as expected
+        self.assertEqual(computed_file.size_in_bytes, 8460)
```

Comment:
```suggestion
        self.assertEqualWithVariance(computed_file.size_in_bytes, 8460, 50)
        
        # outside of this test
        def assertEqualWithVariance(self, value, expected, variance):
            self.assertGreater(expected + variance, value)
            self.assertLess(expected - variance, value)
        
```

If you find that the value changes and you cannot reliably assert that the value matches you can create a test that allows for some wiggle room.

Reply:
This is such a great alternative! I've applied it at daccc79a58a3a61df45f50277bc18a8368e21432 👍 

I’ve also added a new method called `assertFields`, which asserts that the fields in the portal metadata computed file match the download configuration settings. We talked about this optimization idea during our 1:1, so I went ahead and applied the input as well!